janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/rest/RestElasticSearchClient.java

```diff
@@ -356,17 +361,22 @@ public void deleteIndex(String indexName) throws IOException {
     public void bulkRequest(List<ElasticSearchMutation> requests, String ingestPipeline) throws IOException {
         final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
         for (final ElasticSearchMutation request : requests) {
-            final Map<String, String> requestData;
-            if(useMappingTypes){
-                requestData = ImmutableMap.of(
+            Map<String, Object> requestData = new HashMap<>();
+            if (useMappingTypes) {
+                requestData.putAll(ImmutableMap.of(
                     "_index", request.getIndex(),
                     "_type", request.getType(),
-                    "_id", request.getId());
+                    "_id", request.getId()));
             } else {
-                requestData = ImmutableMap.of(
+                requestData.putAll(ImmutableMap.of(
                     "_index", request.getIndex(),
-                    "_id", request.getId());
+                    "_id", request.getId()));
             }
+
+            if (retryOnConflict != null && request.getRequestType() == ElasticSearchMutation.RequestType.UPDATE) {
+                requestData.putAll(ImmutableMap.of(retryOnConflictKey, retryOnConflict));
```

Comment:
We can ommit additional map creation here by using 'put' method instead of putAll.
'requestData.put(retryOnConflictKey, retryOnConflict);'

Reply:
updated